featurebyte/feast/patch.py

```diff
@@ -101,93 +139,6 @@ def augment_response_with_on_demand_transforms(
-        alias: str
-            Alias name of the column
-        """
-        self.attrs["_alias"][alias] = column_name
-
-    def __getitem__(self, key: Any) -> Union[pd.Series, pd.DataFrame]:
-        if not isinstance(key, str) and isinstance(key, Iterable):
-            return pd.DataFrame({_key: self.__getitem__(_key) for _key in key})
-
-        if isinstance(key, str) and key in self.attrs["_alias"]:
-            key = self.attrs["_alias"][key]
-
-        return super().__getitem__(key)
-
-
-def get_transformed_features_df(
-    feature_view: OnDemandFeatureView,
-    df_with_features: pd.DataFrame,
-    full_feature_names: bool = False,
-) -> pd.DataFrame:
-    """
-    The main difference between this and the original Feast implementation is that an extended pandas
-    DataFrame with alias column names support is used to improve the runtime & memory performance of the
-    function.
-
-    Parameters
-    ----------
-    feature_view: OnDemandFeatureView
-        OnDemandFeatureView object
-    df_with_features: pd.DataFrame
-        Dataframe with features
-    full_feature_names: bool
-        A boolean that provides the option to add the feature view prefixes to the feature names,
-
-    Returns
-    -------
-    pd.DataFrame
-        Dataframe with transformed features
-    """
-    # Original implementation assigns a new column on each iteration, this implementation uses a wrapper
-    # class to improve the runtime & memory performance.
-    df_with_features = DataFrameWrapper(df_with_features)
-    for source_fv_projection in feature_view.source_feature_view_projections.values():
-        for feature in source_fv_projection.features:
-            full_feature_ref = f"{source_fv_projection.name}__{feature.name}"
-            if full_feature_ref in df_with_features.keys():
-                # Make sure the partial feature name is always present
-                df_with_features.add_column_alias(full_feature_ref, feature.name)
-            elif feature.name in df_with_features.keys():
-                # Make sure the full feature name is always present
-                df_with_features.add_column_alias(feature.name, full_feature_ref)
-
-    # Compute transformed values and apply to each result row
-    df_with_transformed_features = feature_view.feature_transformation.udf(df_with_features)  # type: ignore
-
-    # Work out whether the correct columns names are used.
-    rename_columns: Dict[str, str] = {}
-    for feature in feature_view.features:
-        short_name = feature.name
-        long_name = f"{feature_view.projection.name_to_use()}__{feature.name}"
-        if short_name in df_with_transformed_features.columns and full_feature_names:
-            rename_columns[short_name] = long_name
-        elif not full_feature_names:
-            # Long name must be in dataframe.
-            rename_columns[long_name] = short_name
-
-    return pd.DataFrame(df_with_transformed_features).rename(columns=rename_columns)
```

Comment:
Newer Feast uses arrow for `pandas` mode:
https://github.com/feast-dev/feast/blob/v0.37.1/sdk/python/feast/on_demand_feature_view.py#L395-L447

This patch becomes irrelevant at this moment.